lms/djangoapps/experiments/views.py

```diff
@@ -46,7 +46,6 @@ def create_or_update(self, request, *args, **kwargs):
             try:
                 obj = self.get_queryset().get(user=self.request.user, experiment_id=experiment_id, key=key)
                 self.kwargs['pk'] = obj.pk
-                self.request.data['id'] = obj.pk
```

Comment:
Was this assignment redundant?

Reply:
Yes, that is my assertion.  From the commit message:

>  avoid unnecessary attempt to update an immutable dict
>
>  request.data has become immutable under Django 1.11 for some reason.
>  Regardless, setting the 'id' key should be extraneous because we already
>  instruct the update method to use a specific pk (see kwargs) which
>  should be sufficient to locate the database record and update it.